components/tag-list/tag-list.js

```diff
@@ -0,0 +1,96 @@
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { ArrowKeysMixin } from '../../mixins/arrow-keys-mixin.js';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+
+const ITEM_SPACING = 6;
+
+class TagList extends ArrowKeysMixin(LitElement) {
+
+	static get properties() {
+		return {
+			/**
+			 * REQUIRED: A description of the tag list for additional accessibility context
+			 * @type {string}
+			 */
+			description: { type: String },
+			/**
+			 * @ignore
+			 */
+			role: { type: String, reflect: true },
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				display: block;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			.tag-list-container {
+				display: flex;
+				flex-wrap: wrap;
+				margin: -${ITEM_SPACING}px -${ITEM_SPACING}px 0 0;
+				padding: 0;
+				position: relative;
+			}
+			::slotted(*) {
+				margin: ${ITEM_SPACING}px ${ITEM_SPACING}px 0 0;
+			}
+		`;
+	}
+
+	constructor() {
+		super();
+		/** @ignore */
+		this.role = 'application';
+		/** @ignore */
+		this.arrowKeysDirection = 'leftrightupdown';
+		this._items = [];
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		requestAnimationFrame(() => {
+			this._items = this._getTagListItems();
+			if (this._items.length > 0) this._items[0].setAttribute('tabindex', 0);
+		});
+	}
+
+	render() {
+		const description = this.description ? html`<div id="d2l-tag-list-description" hidden>${this.description}</div>` : null;
```

Comment:
Since `description` is required, you could simplify this to just always render it.

Reply:
My worry was people might still forget to set it and then I could degrade gracefully, but I guess if we saw that happening we could decide to start throwing an error or something similar.  Will update!

Reply:
Yeah or throw an exception after a second or so if it's still unset like we do with input labels.